### PR TITLE
Adding esbuild bundler to the pipeline

### LINKED ISSUE #3377: ci: Add esbuild in package pipeline
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Following: https://github.com/fastify/fastify/issues/3371

Would be nice to have `esbuild` in the CI for bundlers.

### PR DESCRIPTION
**Pull request by jhonrocha**
This PR fixes #3377: It includes esbuild tests to the package pipeline.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/bundler/esbuild/bundler-test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const fastifySuccess = require('./dist/success')
+const fastifyFailPlugin = require('./dist/failPlugin')
+
+test('Bundled package should work', t => {
+  t.plan(1)
+  fastifySuccess.ready((err) => {
```
Comment (RafaelGSS) [inline test/bundler/esbuild/bundler-test.js:10]: Use the `inject` to test if the route was registered with success

File: test/bundler/esbuild/src/fail-plugin-version.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const fp = require('fastify-plugin')
+const fastify = require('../../../../')({
+  logger: true
```
Comment (RafaelGSS) [inline test/bundler/esbuild/src/fail-plugin-version.js:3]: The logger in this case is not necessary.

**Review by RafaelGSS — COMMENTED** [review]
Few nitpicks

**Review by RafaelGSS — APPROVED** [review]
LGTM.

File: test/bundler/esbuild/bundler-test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        t.same(res.json(), { hello: 'world' })
+      }
+    )
+  })
+})
+
+test('Bundled package should not work with bad plugin version', (t) => {
+  t.plan(1)
+  fastifyFailPlugin.ready((err) => {
+    t.ok(err)
```
Comment (Eomm) [inline test/bundler/esbuild/bundler-test.js:29]: I would check using `t.match(err.message, /something/i)` to be sure that the error is the one expected.

It happened that an error has changed in past and we did not spot it

**Review by Eomm — APPROVED** [review]

**Pull request merged**